pkgs/controllers/certificaterequest.go

```diff
@@ -50,6 +55,55 @@ func (r *CertificateRequestController) Reconcile(req reconcile.Request) (reconci
 		return reconcile.Result{}, nil
 	}
 
+	// Ignore CertificateRequest if it is already Ready
+	if cmutil.CertificateRequestHasCondition(cr, certmanager.CertificateRequestCondition{
```

Comment:
I have two questions:

1. Since this code is now checking the Ready condition (and `cr.Status.FailureTime`), do I still need to check for `cr.Status.Certificate`? I'm still drinking my morning coffee, but it seems to me that `Ready==True` could only be possible if `cr.Status.Certificate` was set.
2. Does it make sense to move some or all of these checks into `cmutil`? Making one call into the `cmutil` package seems better than chaining multiple together checks here.

Reply:
1. We shouldn't _have_ to, but we do anyway to make sure we cover any cases where another signer has behaved improperly. This makes the code as safe as possible.

2. Yes, we could think about moving these things to a chain in cert-manager utils. This would definitely help out with code duplication across projects. We would loose the specific logging for each case however.

Reply:
The utility function could return an error with that detailed information, if it's found to be useful to logging. In the meantime, this is fine.